test/test.xunit.execution/Sdk/Frameworks/Runners/XunitTestCollectionRunnerTests.cs

```diff
@@ -38,6 +38,58 @@ public class XunitTestCollectionRunnerTests
         Assert.True(fixtureUnderTest.Disposed);
     }
 
+    [Fact]
+    public static async void DisposeAndAsyncLifetimeShouldBeCalledInTheRightOrder()
+    {
+        var collection = new TestCollection(Mocks.TestAssembly(), Reflector.Wrap(typeof(CollectionForFixtureAsyncLifetimeAndDisposableUnderTest)), null);
+        var testCase = Mocks.XunitTestCase<XunitTestCollectionRunnerTests>("DisposeAndAsyncLifetimeShouldBeCalledInTheRightOrder", collection);
+        var runner = TestableXunitTestCollectionRunner.Create(testCase);
+
+        var runnerSessionTask = runner.RunAsync();
+
+        await Task.Delay(500);
+
+        var fixtureUnderTest = runner.CollectionFixtureMappings.Values.OfType<FixtureAsyncLifetimeAndDisposableUnderTest>().Single();
+
+        Assert.True(fixtureUnderTest.DisposeAsyncCalled);
+        Assert.False(fixtureUnderTest.Disposed);
+
+        fixtureUnderTest.DisposeAsyncSignaler.SetResult(true);
+
+        await runnerSessionTask;
+
+        Assert.True(fixtureUnderTest.Disposed);
+    }
+
+    class CollectionForFixtureAsyncLifetimeAndDisposableUnderTest : ICollectionFixture<FixtureAsyncLifetimeAndDisposableUnderTest> { }
+
+    class FixtureAsyncLifetimeAndDisposableUnderTest : IAsyncLifetime, IDisposable
+    {
+        public bool Disposed;
+
+        public bool DisposeAsyncCalled { get; private set; }
```

Comment:
Why one a property but other a field? (fields are fine here IMO)

Reply:
Just a habit. Sorry, did not follow the code. Fixing it now.